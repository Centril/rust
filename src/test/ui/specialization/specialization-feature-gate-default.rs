// Check that specialization must be ungated to use the `default` keyword

// gate-test-specialization

trait Foo {
    fn foo(&self);
}

#[cfg(FALSE)]
impl<T> Foo for T {
    default //~ ERROR specialization is unstable
    fn foo(&self) {}
}

#[cfg(FALSE)]
default //~ ERROR specialization is unstable
impl<T> Foo for T {
    fn foo(&self) {}
}

fn main() {}
