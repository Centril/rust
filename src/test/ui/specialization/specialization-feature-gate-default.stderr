error[E0658]: specialization is unstable
  --> $DIR/specialization-feature-gate-default.rs:11:5
   |
LL |     default
   |     ^^^^^^^
   |
   = note: for more information, see https://github.com/rust-lang/rust/issues/31844
   = help: add `#![feature(specialization)]` to the crate attributes to enable

error[E0658]: specialization is unstable
  --> $DIR/specialization-feature-gate-default.rs:16:1
   |
LL | default
   | ^^^^^^^
   |
   = note: for more information, see https://github.com/rust-lang/rust/issues/31844
   = help: add `#![feature(specialization)]` to the crate attributes to enable

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0658`.
